apps/studio/components/ui/AIAssistantPanel/MessageMarkdown.tsx

```diff
@@ -50,6 +64,58 @@ export const InlineCode = memo(
 )
 InlineCode.displayName = 'InlineCode'
 
+export const Hyperlink = memo(({ href, children }: { href?: string; children: ReactNode }) => {
+  const isExternalURL = !href?.startsWith('https://supabase.com/dashboard')
+
+  return (
+    <Dialog>
+      <DialogTrigger asChild>
+        <span
+          className={cn(
+            '!m-0 text-foreground cursor-pointer transition',
+            'underline underline-offset-2 decoration-foreground-muted hover:decoration-foreground-lighter'
+          )}
+        >
+          {children}
+        </span>
+      </DialogTrigger>
+      <DialogContent size="small">
+        <DialogHeader className="border-b">
+          <DialogTitle>Verify the link before navigating</DialogTitle>
+        </DialogHeader>
+
+        <DialogSection className="flex flex-col">
+          <p className="text-sm text-foreground-light">
+            This link will take you to the following URL:
+          </p>
+          <p className="text-sm text-foreground">{href}</p>
+          <p className="text-sm text-foreground-light mt-2">Are you sure you want to head there?</p>
+        </DialogSection>
+
+        <DialogFooter>
+          <DialogClose asChild>
+            <Button type="default" className="opacity-100">
+              Cancel
+            </Button>
+          </DialogClose>
+          <DialogClose asChild>
+            <Button asChild type="primary" className="opacity-100">
+              <a
+                href={href}
```

Comment:
should we do same safety transformation as Markdown React? https://github.com/remarkjs/react-markdown/blob/fda7fa560bec901a6103e195f9b1979dab543b17/lib/index.js#L425

Reply:
i'll add an additional state here if an unsafe URL is detected based on ^ this safety transformation
![image](https://github.com/user-attachments/assets/33ae2243-c587-464f-8b08-15f6f99badea)

Reply:
okay yeap makes sense - will render raw text if the URL is not safe then

Reply:
just pushed up the changes! 🙏